src/java.base/share/classes/sun/security/util/DerInputBuffer.java

```diff
@@ -61,13 +61,12 @@
     }
 
     DerInputBuffer dup() {
-        try {
-            DerInputBuffer retval = (DerInputBuffer)clone();
-            retval.mark(Integer.MAX_VALUE);
-            return retval;
-        } catch (CloneNotSupportedException e) {
-            throw new IllegalArgumentException(e.toString());
-        }
+        DerInputBuffer retval = new DerInputBuffer(this.buf);
+        retval.pos = this.pos;
+        retval.count = this.count;
+        retval.allowBER = this.allowBER;
+        retval.mark = this.pos;
+        return retval;
```

Comment:
This is same as calling clone()? If so, not seeing the benefit of having dup()...

Reply:
There is an extra `mark` call but seems not used anywhere.